sdk/translation/ai-translation-document-rest/samples-dev/batchDocumentTranslation.ts

```diff
@@ -1,46 +1,118 @@
+ * It requires the following environment variables to be set:
+ * - DOCUMENT_TRANSLATION_ENDPOINT: The endpoint URL for the Document Translation service.
+ * - STORAGE_BLOB_ENDPOINT: The endpoint URL for the Azure Blob Storage account.
+ * - TRANSLATION_FILE: The URL of the PDF file to be translated.
+ */
+
 import "dotenv/config";
-import createClient from "../src/documentTranslationClient.js";
-import {
-  ONE_TEST_DOCUMENTS,
-  createSourceContainer,
-  createTargetContainer,
-} from "../test/public/utils/samplesHelper.js";
-import {
-  createSourceInput,
-  createTargetInput,
-  createBatchRequest,
-  getTranslationOperationID,
-} from "../test/public/utils/testHelper.js";
-import { isUnexpected } from "../src/isUnexpected.js";
+import createClient, {
+  getLongRunningPoller,
+  isUnexpected,
+} from "@azure-rest/ai-translation-document";
+import { DefaultAzureCredential } from "@azure/identity";
+import { BlobServiceClient } from "@azure/storage-blob";
+import { createRestError } from "@azure-rest/core-client";
+import https from "node:https";
+import path from "node:path";
+import fs from "node:fs";
+import { fileURLToPath } from "node:url";
+
 const endpoint =
-  process.env["ENDPOINT"] ||
+  process.env["DOCUMENT_TRANSLATION_ENDPOINT"] ||
   "https://<translator-instance>-doctranslation.cognitiveservices.azure.com";
-const apiKey = process.env["DOCUMENT_TRANSLATION_API_KEY"] || "<API_Key>";
-const credentials = { key: apiKey ?? "" };
+const blobEndpoint =
+  process.env["STORAGE_BLOB_ENDPOINT"] || "https://<storage-account-name>.blob.core.windows.net";
+const fileUrl =
+  process.env["TRANSLATION_FILE"] || "https://constitutioncenter.org/media/files/constitution.pdf";
 
 export async function main(): Promise<void> {
   console.log("== Batch Document Translation ==");
-  const client = createClient(endpoint, credentials);
+  const credential = new DefaultAzureCredential();
+  const client = createClient(endpoint, credential);
+  const blobServiceClient = new BlobServiceClient(blobEndpoint, credential);
+  const sourceContainerClient = blobServiceClient.getContainerClient("docs");
+  await sourceContainerClient.createIfNotExists();
+  const sourceBlobName = "file.pdf";
+  const sourceBlobClient = sourceContainerClient.getBlobClient(sourceBlobName);
+  const blockBlobClient = sourceBlobClient.getBlockBlobClient();
 
-  const sourceUrl = await createSourceContainer(ONE_TEST_DOCUMENTS);
-  const sourceInput = createSourceInput(sourceUrl);
-  const targetUrl = await createTargetContainer();
-  const targetInput = createTargetInput(targetUrl, "fr");
-  const batchRequest = createBatchRequest(sourceInput, [targetInput]);
+  await new Promise<void>((resolve, reject) => {
+    https
+      .get(fileUrl, (response) => {
+        if (response.statusCode !== 200) {
+          reject(new Error(`Failed to fetch file. Status code: ${response.statusCode}`));
+          return;
+        }
+        blockBlobClient
```

Comment:
no other ways to avoid having to create a `Promise`?  Could we resolve the stream and move blob operation out of here so we can use `await`?

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-js/pull/33640/commits/18ae14fd1dd8b4ebdd54c26850cf62a9b3db536a